From c02d556ab2ea4ceb86009065e33b76556a49c84b Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Sat, 4 Apr 2020 15:23:45 +0800
Subject: [PATCH] hwrng: core - Call hwrng_modinit() early when
 CONFIG_ROCKCHIP_THUNDER_BOOT=y

hw_random driver depends on hwrng_modinit() to init first.
But driver initcall and hwrng_modinit() are both in the same
device_init level.

Fix this by move hwrng_modinit() to the rootfs_initcall level.

Change-Id: I501408d94b7a91fb439512ecc51f20453c5f2382
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 drivers/char/hw_random/core.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/char/hw_random/core.c b/drivers/char/hw_random/core.c
index 0ef7cb0448e8..7716fc4cfcdd 100644
--- a/drivers/char/hw_random/core.c
+++ b/drivers/char/hw_random/core.c
@@ -617,7 +617,11 @@ static void __exit hwrng_modexit(void)
 	unregister_miscdev();
 }
 
+#ifdef CONFIG_ROCKCHIP_THUNDER_BOOT
+rootfs_initcall(hwrng_modinit);
+#else
 module_init(hwrng_modinit);
+#endif
 module_exit(hwrng_modexit);
 
 MODULE_DESCRIPTION("H/W Random Number Generator (RNG) driver");
-- 
2.35.3

